public/app/features/explore/utils/links.ts

```diff
@@ -133,14 +136,24 @@ export const getFieldLinksForExplore = (options: {
           variableMap = variableMapData.variableMap;
         }
 
+        const splitFnWithTracking = (options?: SplitOpenOptions<DataQuery>) => {
+          if (link.origin === DataLinkConfigOrigin.Correlations) {
```

Comment:
When I think about it again it might be actually useful to track all clicks and use origin as a param + add `app` so it's easier to add more tracking in the future like:

```typescript
reportInteraction('grafana_data_link_clicked', {
    origin: link.origin || DataLinkOrigin.Datasource,
    app: CoreApp.Explore,
    internal: true,
  });
```

With this in place I was thinking that adding clicks on external links can also be helpful:

```typescript
linkModel.onClick = (...args) => {
  reportInteraction('grafana_data_link_clicked', {
    origin: link.origin || DataLinkOrigin.Datasource,
    app: CoreApp.Explore,
    internal: false,
  });
  linkModel.onClick?.apply(...args);
};
```